l1-contracts/contracts/state-transition/chain-deps/facets/Executor.sol

```diff
@@ -515,14 +515,17 @@ contract ExecutorFacet is ZKChainBase, IExecutor {
         s.totalBatchesVerified = currentTotalBatchesVerified;
     }
 
-    function _verifyProof(uint256[] memory proofPublicInput, uint256[] memory _proof) internal view {
+    function _verifyProof(uint256[] memory proofPublicInput, uint256[] memory _proof) internal {
         // We can only process 1 batch proof at a time.
         if (proofPublicInput.length != 1) {
             revert CanOnlyProcessOneBatch();
         }
 
-        bool successVerifyProof = s.verifier.verify(proofPublicInput, _proof);
-        if (!successVerifyProof) {
+        (bool callSuccess, bytes memory successVerifyProof) = address(s.dualVerifier).delegatecall(abi.encodeWithSelector(s.dualVerifier.verify.selector, proofPublicInput, _proof));
```

Comment:
No delegate calls please! Why it can't be simple `staticcall` that provides some clear guaruntee like not changing the state?

Reply:
I had the idea of accessing ZKChainStorage state from the DualVerifier as a delegatecall.